### DISCUSSION TITLE
No create id or Any other information in Post method of default PostgREST  APIs

### DISCUSSION THREAD

**Discussion by nithin-pb** [supabase/supabase#2569] [Questions] [answered]
Hi,
I am using superbase with default  PostgREST Apis. but the **APIs don't return any id in response to create operation (post-operation).**

I tried creating stored procedure in superbase for the same and i returned create id in sp response. but that seems to be a hack and it seems to be difficult to create sp for all tables. Is there any alternative for this issue, i mean any configuration or setting where i can configure the response structure of those APIs?

**Comment by steve-chavez** [comment, marked as answer]
Hey @nithinpachday,

> I am using superbase with default PostgREST Apis. but the APIs don't return any id in response to create operation (post-operation)

You need to use the `Prefer: return=representation` header for returning data after a POST.

For more details, see: https://postgrest.org/en/stable/api.html#insertions-updates

**Reply by nithin-pb** [reply]
Thanks, @steve-chavez. Working fine after adding the header
